src/Shared/Debugging/DebugUtils.cs

```diff
@@ -22,13 +23,27 @@ private enum NodeMode
 
         static DebugUtils()
         {
-            string environmentDebugPath = Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH");
-            var debugDirectory = environmentDebugPath;
+            string environmentDebugPath = FileUtilities.TrimAndStripAnyQuotes(Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH"));
+            string debugDirectory = environmentDebugPath;
 
             if (Traits.Instance.DebugEngine)
             {
                 debugDirectory ??= Path.Combine(Directory.GetCurrentDirectory(), "MSBuild_Logs");
 
+                // Probe writeability
+                try
+                {
+                    debugDirectory = ProbeWriteability(debugDirectory);
+                }
+                catch (ArgumentException)
+                {
+                    // This can happen if MSBUILDDEBUGPATH contains invalid characters, but the current working directory may still work.
+                    debugDirectory = ProbeWriteability(Path.Combine(Directory.GetCurrentDirectory(), "MSBuild_Logs"));
+                }
+
+                // Redirect to TEMP if we failed to write to either MSBUILDDEBUGPATH or the current working directory.
+                debugDirectory ??= Path.Combine(Path.GetTempPath(), "MSBuild_Logs");
```

Comment:
```suggestion
                debugDirectory = ProbeWriteability(debugDirectory);
                debugDirectory ??= ProbeWriteability(Path.Combine(Directory.GetCurrentDirectory(), "MSBuild_Logs"));
                debugDirectory ??= Path.Combine(Path.GetTempPath(), "MSBuild_Logs");
```

If `ProbeWriteability` can be changed to catch its own ArgumentException and return null in that case, it simplifies the logic here.

Reply:
I was trying to get out better perf, but on reflection, that isn't a very satisfying reason for making this more complicated. This runs once, and the difference is tiny, so I really shouldn't care. I'll change this.